sql-native/src/main/java/org/ballerinalang/sql/datasource/SQLDatasource.java

```diff
@@ -406,7 +406,13 @@ private AtomikosDataSourceBean buildXAAwareDataSource(SQLDatasourceParams sqlDat
     }
 
     private String buildErrorMessage (Throwable t) {
-        StringBuilder message = new StringBuilder("Error in SQL connector configuration: " + t.getMessage() + "");
+        String msg = t.getMessage();
+        if (msg.contains("ClassNotFoundException")) {
```

Comment:
can't we check whether the throwable instance is `ClassNotFoundException`

Reply:
No.  
This returns the `RuntimeException` and message contains `java.lang.ClassNotFoundException`. So we can't use `instanceof` condition.

Reply:
Can you copy full stacktrace here

Reply:
Please find the following:

**Exception in string:** `java.lang.RuntimeException: java.lang.ClassNotFoundException: com.mysql.cj.jdbc.MysqlDataSource`

**Stacktrace:** ```[com.zaxxer.hikari.util.UtilityElf.createInstance(UtilityElf.java:113), com.zaxxer.hikari.pool.PoolBase.initializeDataSource(PoolBase.java:316), com.zaxxer.hikari.pool.PoolBase.<init>(PoolBase.java:109), com.zaxxer.hikari.pool.HikariPool.<init>(HikariPool.java:108), com.zaxxer.hikari.HikariDataSource.<init>(HikariDataSource.java:81), org.ballerinalang.sql.datasource.SQLDatasource.buildNonXADataSource(SQLDatasource.java:353), org.ballerinalang.sql.datasource.SQLDatasource.<init>(SQLDatasource.java:81), org.ballerinalang.sql.datasource.SQLDatasource.createAndInitDatasource(SQLDatasource.java:237), org.ballerinalang.sql.datasource.SQLDatasource.lambda$retrieveDatasource$1(SQLDatasource.java:151), java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1705), org.ballerinalang.sql.datasource.SQLDatasource.retrieveDatasource(SQLDatasource.java:150), org.ballerinalang.sql.nativeimpl.ClientProcessor.createClient(ClientProcessor.java:58), org.ballerinalang.mysql.nativeimpl.ClientProcessor.createClient(ClientProcessor.java:80), ballerinax.mysql.0_7_0-alpha8.client.createClient(client.bal:180), ballerinax.mysql.0_7_0-alpha8.$value$Client.init(client.bal:48), ballerinax.mysql.0_7_0-alpha8.$value$Client.call(client.bal), ballerinax.mysql.0_7_0-alpha8.$value$Client.$init$(client.bal:22), ballerinax.mysql.0_7_0-alpha8.$value$Client.call(client.bal), kalaiyarasiganeshalingam.jdbc.0_1_0.$_init.$gen$$0046$0060init$0062(jdbc:10), kalaiyarasiganeshalingam.jdbc.0_1_0.$_init.$moduleInit(jdbc), kalaiyarasiganeshalingam.jdbc.0_1_0.$_init.$lambda$$moduleInit$(jdbc), io.ballerina.runtime.internal.scheduling.SchedulerItem.execute(Scheduler.java:571), io.ballerina.runtime.internal.scheduling.Scheduler.run(Scheduler.java:301), io.ballerina.runtime.internal.scheduling.Scheduler.runSafely(Scheduler.java:269), java.base/java.lang.Thread.run(Thread.java:834)]```

Reply:
Okay, can we check instead of with `t.getCause()` ?

Reply:
Fixed in bf2b23f